ERROR: Unsupported distribution 'arch'

### Description

DreamServer fails to install

### Steps to Reproduce

1. `./install.sh`
2. Select option 1 for full installation

### Expected Behavior

I expect installation to complete successfully

### Actual Behavior

I see the following output

```
# Executing docker install script, commit: f381ee68b32e515bb4dc034b339266aff1fbc460

ERROR: Unsupported distribution 'arch'

[ERROR] Docker installation failed. Check network connectivity and try again.
```

### Operating System

Arch Linux

### GPU

None (CPU Only)

### Docker Version

N/A

### VRAM

N/A

### Logs

```shell
[0;36mDream Server Installer[0m

[0;32m[INFO][0m Detected distro: arch (like: none, pkg: pacman)
[H[2J
[1;32m    ____                                 _____[0m
[1;32m   / __ \ _____ ___   ____ _ ____ ___   / ___/ ___   _____ _   __ ___   _____[0m
[1;32m  / / / // ___// _ \ / __ `// __ `__ \  \__ \ / _ \ / ___/| | / // _ \ / ___/[0m
[1;32m / /_/ // /   /  __// /_/ // / / / / / ___/ //  __// /    | |/ //  __// /[0m
[1;32m/_____//_/    \___/ \__,_//_/ /_/ /_/ /____/ \___//_/     |___/ \___//_/[0m

  █░▒█▓█▒░▒█▒░▒▒▒▓▓█▒▓▓▓▒▓▒▓▒▒▓██▒▓▓█▓█▓░▓█▓▒██░▓█░░░█░▓░▒░░▓▓░▓▓
[1;32m  D R E A M G A T E[0m   [0;32mLocal AI // Sovereign Intelligence // 2026[0m
[2;32m  CLASSIFICATION: FREEDOM IMMINENT[0m
[2;32m  BUILD: v2.3.2 // 2026-03-21 10:10[0m
  █▒░░█▓█░█▓░▒▓░░▓▒▓▓░▒░▓███▓▒▒░███▒█▒▒▓░▓░▒░░▓░░▒█░█▒▓▓▒█▒▓░▒░█▒


.
..
...
   
[0;32mS█i█g█n█a█l█ █a█c█q█u█i█r█e█d█.[0m
[0;32mI█ █w█i█l█l█ █g█u█i█d█e█ █t█h█e█ █i█n█s█t█a█l█l█a█t█i█o█n█.█ █S█t█a█y█ █w█i█t█h█ █m█e█.[0m

  [0;33mVersion 2.3.2[0m

[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[0;32mTip:[0m Press Ctrl+C twice to abort.
[0;32m──────────────────────────────────────────────────────────────────────────────[0m


[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[1;32mDREAMGATE SEQUENCE [10:10:34][0m  [0;32mPHASE 1/6 — Pre-flight Checks[0m
[0;33mEST. TIME:[0m ~30 seconds
[0;32m──────────────────────────────────────────────────────────────────────────────[0m
  [0;32m▸[0m I'm scanning your system for required components...
[0;32m[INFO][0m Detected OS: Arch Linux
[0;32m[INFO][0m curl: curl 8.19.0 (x86_64-pc-linux-gnu) libcurl/8.19.0 OpenSSL/3.6.1 zlib/1.3.2 brotli/1.2.0 zstd/1.5.7 libidn2/2.3.8 libpsl/0.21.5 libssh2/1.11.1 nghttp2/1.68.1 ngtcp2/1.21.0 nghttp3/1.15.0 mit-krb5/1.21.3
[0;33m[WARN][0m Optional tools missing: rsync
  These are needed for update/backup scripts. Install with:
  sudo pacman -S rsync
  [1;32m✓[0m Pre-flight checks passed.
  [0;32m░▒▓█▓▒░[0m No cloud dependencies required for core operation.

[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[1;32mSYSTEM DETECTION[0m
[0;32m──────────────────────────────────────────────────────────────────────────────[0m
  [0;32m▸[0m Reading hardware telemetry...
[0;32m[INFO][0m Capability profile loaded: /tmp/dream-server-capabilities.json
[0;32m[INFO][0m Capability profile: platform=linux, gpu=none, tier=T1
[0;32m[INFO][0m Hardware class: unknown (Unknown)
[0;32m[INFO][0m RAM: 62GB
[0;32m[INFO][0m Available disk: 3463GB
  [0;32m▸[0m Detecting GPU...
[0;33m[WARN][0m No GPU detected. Falling back to CPU-only mode (inference will be slow).
[0;32m[INFO][0m CPU-only mode: llama.cpp will use CPU inference. Consider adding a GPU for better performance.
[0;32m[INFO][0m Capabilities override detection: backend=nvidia, memory=discrete, tier=T1
[0;32m[INFO][0m Backend contract loaded: /home/jer/Downloads/DreamServer/dream-server/config/backends/cpu.json
[0;32m[INFO][0m Backend runtime: cpu (llama-server)
[0;32m[INFO][0m Auto-detected tier: 1
[0;32m[INFO][0m Compose selection: -f docker-compose.base.yml -f docker-compose.cpu.yml -f extensions/services/ape/compose.yaml -f extensions/services/comfyui/compose.yaml -f extensions/services/comfyui/compose.nvidia.yaml -f extensions/services/dashboard/compose.local.yaml -f extensions/services/embeddings/compose.yaml -f extensions/services/litellm/compose.yaml -f extensions/services/litellm/compose.local.yaml -f extensions/services/n8n/compose.yaml -f extensions/services/open-webui/compose.local.yaml -f extensions/services/openclaw/compose.yaml -f extensions/services/openclaw/compose.local.yaml -f extensions/services/perplexica/compose.yaml -f extensions/services/perplexica/compose.local.yaml -f extensions/services/privacy-shield/compose.yaml -f extensions/services/privacy-shield/compose.local.yaml -f extensions/services/qdrant/compose.yaml -f extensions/services/searxng/compose.yaml -f extensions/services/token-spy/compose.yaml -f extensions/services/tts/compose.yaml -f extensions/services/whisper/compose.yaml -f extensions/services/whisper/compose.nvidia.yaml

[0;32m+-------------------------------------------------------------+[0m
[0;32m|[0m  [1;32mHARDWARE SCAN RESULTS[0m                                      [0;32m|[0m
[0;32m+-------------------------------------------------------------+[0m
[0;32m|[0m  GPU:    None                                               [0;32m|[0m
[0;32m|[0m  VRAM:   0GB                                                [0;32m|[0m
[0;32m|[0m  CPU:    Intel(R) Core(TM) Ultra 7 155H                     [0;32m|[0m
[0;32m|[0m  RAM:    62GB                                               [0;32m|[0m
[0;32m|[0m  Disk:   3463GB available                                   [0;32m|[0m
[0;32m+-------------------------------------------------------------+[0m

[0;32m+-------------------------------------------------------------+[0m
[0;32m|[0m  [1;32mCLASSIFICATION: TIER 1[0m                                      [0;32m|[0m
[0;32m+-------------------------------------------------------------+[0m
[0;32m|[0m  Model:   qwen3-8b                                          [0;32m|[0m
[0;32m|[0m  Speed:   ~25 tokens/second                                 [0;32m|[0m
[0;32m|[0m  Users:   1-2 concurrent comfortably                        [0;32m|[0m
[0;32m+-------------------------------------------------------------+[0m

[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[1;32mDREAMGATE SEQUENCE [10:10:34][0m  [0;32mPHASE 2/6 — Feature Selection[0m
[0;33mEST. TIME:[0m ~1 minute
[0;32m──────────────────────────────────────────────────────────────────────────────[0m

  [0;32m▸[0m Choose how deep you want to go. I can install everything, or keep it minimal.

  [1;32m[1][0m Full Stack [0;33m(recommended — just press Enter)[0m
      Chat + Voice + Workflows + Document Q&A + AI Agents
      ~16GB download, all features enabled

  [1;32m[2][0m Core Only
      Chat interface + API
      ~12GB download, minimal footprint

  [1;32m[3][0m Custom
      Choose exactly what you want

  Select an option [1]: 
  [0;32m░▒▓█▓▒░[0m Acknowledged.
[0;32m[INFO][0m Selected: Full Stack
[0;32m[INFO][0m OpenClaw config: minimal.json (matched to Tier 1)
[0;32m[INFO][0m All services enabled (core install)

[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[1;32mREQUIREMENTS CHECK[0m
[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[0;32m[INFO][0m Preflight report: /tmp/dream-server-preflight-report.json
  [1;32m✓[0m Preflight passed with 1 warning(s).
  - WARN: NVIDIA backend selected but no NVIDIA GPU VRAM was detected.
    Suggestion: Install/verify NVIDIA drivers or switch to a supported AMD path.

[0;32m──────────────────────────────────────────────────────────────────────────────[0m
[1;32mDREAMGATE SEQUENCE [10:10:36][0m  [0;32mPHASE 3/6 — Docker Setup[0m
[0;33mEST. TIME:[0m ~2 minutes
[0;32m──────────────────────────────────────────────────────────────────────────────[0m
  [0;32m▸[0m Preparing container runtime...
  [0;32m▸[0m Installing Docker...
# Executing docker install script, commit: f381ee68b32e515bb4dc034b339266aff1fbc460

ERROR: Unsupported distribution 'arch'

[0;31m[ERROR][0m Docker installation failed. Check network connectivity and try again.
```

### Screenshots

_No response_

--- a/dream-server/install-core.sh
+++ b/dream-server/install-core.sh
@@ -75,7 +75,12 @@ source "$SCRIPT_DIR/installers/lib/packaging.sh"
 source "$SCRIPT_DIR/installers/lib/progress.sh"
 if [[ -f "$SCRIPT_DIR/lib/service-registry.sh" ]]; then 
     source "$SCRIPT_DIR/lib/service-registry.sh" 
-    sr_load 
+    sr_load
+    if [[ "${_SR_FAILED:-}" == "true" ]]; then
+        warn "Service registry failed to load. Will retry after dependencies are installed."
+        _SR_LOADED=false
+        _SR_FAILED=false
+    fi
 fi
 
 #=============================================================================

--- a/dream-server/installers/lib/packaging.sh
+++ b/dream-server/installers/lib/packaging.sh
@@ -131,12 +131,14 @@ pkg_resolve() {
         apt)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-compose-plugin" ;;
+                python3-pyyaml)        echo "python3-yaml" ;;
                 *) echo "$canonical" ;;
             esac
             ;;
         dnf)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-compose-plugin" ;;
+                python3-pyyaml)        echo "python3-pyyaml" ;;
                 build-essential)       echo "gcc gcc-c++ make" ;;
                 *) echo "$canonical" ;;
             esac
@@ -144,6 +146,7 @@ pkg_resolve() {
         pacman)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-compose" ;;
+                python3-pyyaml)        echo "python-yaml" ;;
                 build-essential)       echo "base-devel" ;;
                 *) echo "$canonical" ;;
             esac
@@ -151,6 +154,7 @@ pkg_resolve() {
         zypper)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-compose" ;;
+                python3-pyyaml)        echo "python3-PyYAML" ;;
                 build-essential)       echo "devel_basis" ;;
                 *) echo "$canonical" ;;
             esac
@@ -158,6 +162,7 @@ pkg_resolve() {
         xbps)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-compose" ;;
+                python3-pyyaml)        echo "python3-yaml" ;;
                 build-essential)       echo "base-devel" ;;
                 *) echo "$canonical" ;;
             esac
@@ -165,6 +170,7 @@ pkg_resolve() {
         apk)
             case "$canonical" in
                 docker-compose-plugin) echo "docker-cli-compose" ;;
+                python3-pyyaml)        echo "py3-yaml" ;;
                 build-essential)       echo "build-base" ;;
                 *) echo "$canonical" ;;
             esac

--- a/dream-server/installers/phases/05-docker.sh
+++ b/dream-server/installers/phases/05-docker.sh
@@ -57,12 +57,42 @@ else
     if $DRY_RUN; then
         log "[DRY RUN] Would install Docker via official script"
     else
-        tmpfile=$(mktemp /tmp/install-docker.XXXXXX.sh)
-        if ! curl -fsSL --max-time 300 https://get.docker.com -o "$tmpfile" || ! sh "$tmpfile"; then
-            rm -f "$tmpfile"
-            error "Docker installation failed. Check network connectivity and try again."
-        fi
-        rm -f "$tmpfile"
+        case "$PKG_MANAGER" in
+            apt|dnf|zypper)
+                # Docker CE via get.docker.com (supports Debian/Ubuntu/Fedora/RHEL/SLES)
+                tmpfile=$(mktemp /tmp/install-docker.XXXXXX.sh)
+                if ! curl -fsSL --max-time 300 https://get.docker.com -o "$tmpfile" || ! sh "$tmpfile"; then
+                    rm -f "$tmpfile"
+                    error "Docker installation failed. Check network connectivity and try again."
+                fi
+                rm -f "$tmpfile"
+                ;;
+            pacman)
+                # Arch/Manjaro/CachyOS/EndeavourOS -- Docker is in the official repos
+                pkg_install docker
+                sudo systemctl enable --now docker.service 2>>"$LOG_FILE" || true
+                ;;
+            xbps)
+                # Void Linux -- runit service management
+                pkg_install docker
+                sudo ln -s /etc/sv/docker /var/service/ 2>/dev/null || true
+                ;;
+            apk)
+                # Alpine -- OpenRC service management
+                pkg_install docker
+                sudo rc-update add docker boot 2>>"$LOG_FILE" || true
+                sudo service docker start 2>>"$LOG_FILE" || true
+                ;;
+            *)
+                # Unknown distro -- try get.docker.com as best-effort fallback
+                tmpfile=$(mktemp /tmp/install-docker.XXXXXX.sh)
+                if ! curl -fsSL --max-time 300 https://get.docker.com -o "$tmpfile" || ! sh "$tmpfile"; then
+                    rm -f "$tmpfile"
+                    error "Docker installation failed. Your distro (${DISTRO_ID:-unknown}) may not be supported. Install Docker manually and re-run with --skip-docker."
+                fi
+                rm -f "$tmpfile"
+                ;;
+        esac
 
         # Add the invoking user (not root) to the docker group
         target_user="${SUDO_USER:-$USER}"

--- a/dream-server/lib/service-registry.sh
+++ b/dream-server/lib/service-registry.sh
@@ -4,6 +4,7 @@
 
 EXTENSIONS_DIR="${SCRIPT_DIR:-$(pwd)}/extensions/services"
 _SR_LOADED=false
+_SR_FAILED=false
 _SR_CACHE="/tmp/dream-service-registry.$$.sh"
 
 # Caching for compose flags (session-level)
@@ -43,7 +44,24 @@ sr_load() {
         PYTHON_CMD="python"
     fi
 
-    "$PYTHON_CMD" - "$EXTENSIONS_DIR" <<'PYEOF' > "$_SR_CACHE"
+    # Ensure PyYAML is available (Arch, Void, Alpine don't ship it by default)
+    if ! "$PYTHON_CMD" -c "import yaml" 2>/dev/null; then
+        if declare -f pkg_install &>/dev/null && declare -f pkg_resolve &>/dev/null; then
+            [[ -z "${PKG_MANAGER:-}" ]] && declare -f detect_pkg_manager &>/dev/null && detect_pkg_manager
+            declare -f log &>/dev/null && log "PyYAML not found; installing system package..."
+            # shellcheck disable=SC2046
+            pkg_install $(pkg_resolve python3-pyyaml) 2>>"${LOG_FILE:-/dev/null}" || true
+        fi
+        if ! "$PYTHON_CMD" -c "import yaml" 2>/dev/null; then
+            declare -f warn &>/dev/null && warn "PyYAML not available. Service registry will be incomplete."
+            declare -f warn &>/dev/null && warn "Install manually: pip3 install pyyaml"
+            _SR_LOADED=true  # Prevent repeated retries
+            _SR_FAILED=true
+            return 0
+        fi
+    fi
+
+    if ! "$PYTHON_CMD" - "$EXTENSIONS_DIR" <<'PYEOF' > "$_SR_CACHE"
 import yaml, sys, os
 from pathlib import Path
 
@@ -151,6 +169,13 @@ for service_dir in sorted(ext_dir.iterdir()):
         print(f'# ERROR: failed to parse {manifest_path}: {exc}', file=sys.stderr)
         continue
 PYEOF
+    then
+        rm -f "$_SR_CACHE"
+        declare -f warn &>/dev/null && warn "Service registry: manifest parser failed"
+        _SR_LOADED=true  # Prevent repeated retries
+        _SR_FAILED=true
+        return 0
+    fi
 
     # Source the generated registry (one subprocess for all manifests)
     [[ -f "$_SR_CACHE" ]] && . "$_SR_CACHE"

